### PR TITLE
Initial React commit for Sorting + Search

### PR DESCRIPTION
**Pull request by NJaryal**
Initial Commit - Please focus on functionality as of now.

UI will be updated as per mockup soon.

### CODE REVIEW

File: src/components/Header.js [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+            return b.vote_count - a.vote_count
+        })
+        this.setState({
+            results: sortResult
+        })
+    }
+    render() {  
+        const data = this.state.results;
+        return (
+            <ErrorBoundary>
```
Comment (byslavik) [inline src/components/Header.js]: You can wrap the whole app with ErrorBoundary

File: src/components/Header.js [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    sortRelease = () => {
+        const movieReleaseDate = this.state.results
+        let sortResult = movieReleaseDate.sort((a, b) => {
+            return new Date(b.release_date) - new Date(a.release_date);
+        })
+        this.setState({
+            results: sortResult
+        })
+    }
+    sortRating = () => {
```
Comment (byslavik) [inline src/components/Header.js]: I would suggest to split you header into smaller components. Something like SortComponent, Search and etc

File: src/components/NotFound.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import React, { Component } from 'react'
+class NotFound extends Component {
```
Comment (byslavik) [inline src/components/NotFound.js]: Please create one/two components with functional components

Comment (NJaryal) [reply src/components/NotFound.js]: Added

File: src/components/Header.js (hunk 1 above)
Comment (NJaryal) [reply src/components/Header.js]: Kept single ErrorBoundary

File: src/components/Header.js (hunk 2 above)
Comment (NJaryal) [reply src/components/Header.js]: Added components

File: src/components/NotFound.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import React from 'react'
+
```
Comment (byslavik) [inline src/components/NotFound.js]: In that file I would suggest to declare NoContent components first and then declare NotFound component. I twill be easier to read

File: src/components/NotFound.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+const NotFound = () => {
+    return <NoContent />;
```
Comment (byslavik) [inline src/components/NotFound.js]: Also there is no need to write return if you do not have any processing inside of component
const NotFound = () => <NoContent />

will be fine

File: src/components/NotFound.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+  };
+  
+const NoContent = () => {
+    const data = 'No data found!';
```
Comment (byslavik) [inline src/components/NotFound.js]: If it is not dynamic data I would suggest to move it out of the scope of component

const data = ''

const NoContent = () => <h1>{ data }</h1>

or even better to pass it as a prop
const NoContent = ({ data }) => <h1>{ data }</h1>
<NoContent data={ data }>

File: src/components/Search.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
+import React from 'react'
+function Search (props) {
```
Comment (byslavik) [inline src/components/Search.js]: could be rewritten with arrow function

`const Search = () => `

File: src/components/Search.js [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+    return ( 
+        <form className="navbar-form navbar-left" role="search">
+        <div className="input-group mb-3">
+            <input type="text" className="form-control" placeholder="Enter movie's name" onChange={props.onkeyHandle} />
+            <div className="input-group-append">
+                <button className="btn searchBtn" type="button" onClick={props.resultsQuery}>Search</button>
```
Comment (byslavik) [inline src/components/Search.js]: you can extract your handlers with destructuring

`const Search = ({ resultsQuery, onkeyHandle }) => `

File: src/components/Search.js [hunk 9]
```diff
@@ -0,0 +1,19 @@
+import React from 'react'
+function Search (props) {
+    return ( 
+        <form className="navbar-form navbar-left" role="search">
+        <div className="input-group mb-3">
+            <input type="text" className="form-control" placeholder="Enter movie's name" onChange={props.onkeyHandle} />
```
Comment (byslavik) [inline src/components/Search.js]: please rename onkeyHandle to onKeyHandle

File: src/components/SortByCategory.js [hunk 10]
```diff
@@ -0,0 +1,16 @@
+import React from 'react'
+
+function SortByCategory (props) {
```
Comment (byslavik) [inline src/components/SortByCategory.js]: the same thing for props and arrow component here

File: src/components/SortSearch.js [hunk 11]
```diff
@@ -0,0 +1,15 @@
+import React from 'react'
+
+function SortSearch(props) {
```
Comment (byslavik) [inline src/components/SortSearch.js]: The same thing here

File: src/components/SortSearch.js [hunk 12]
```diff
@@ -0,0 +1,15 @@
...
+    return (
+        <div className="btn-group" role="group" aria-label="Basic example">
+            SEARCH BY 
+            <button type="button" className="btn btn-sort_toggle" onClick={props.sortTitles}>Title</button>
```
Comment (byslavik) [inline src/components/SortSearch.js]: you can create a separate component for button to avoid writing of the the same classNames and etc

Comment (byslavik) [reply src/components/SortSearch.js]: and use it as <Button onClick={ handler } />

File: src/components/NotFound.js (hunk 5 above)
Comment (NJaryal) [reply src/components/NotFound.js]: Updated

File: src/components/NotFound.js (hunk 6 above)
Comment (NJaryal) [reply src/components/NotFound.js]: Updated

File: src/components/Search.js (hunk 7 above)
Comment (NJaryal) [reply src/components/Search.js]: Arrow Functionality implemented

File: src/components/Search.js (hunk 9 above)
Comment (NJaryal) [reply src/components/Search.js]: Renamed

File: src/components/SortByCategory.js (hunk 10 above)
Comment (NJaryal) [reply src/components/SortByCategory.js]: Arrow Functionality Implemented

File: src/components/Main.js [hunk 13]
```diff
@@ -0,0 +1,11 @@
+import React, { Component } from 'react'
+import Header from './Header'
+class Main extends Component {
```
Comment (byslavik) [inline src/components/Main.js]: If you dont have any logic inside of component, you can use functional components

File: src/components/MoviesList.js [hunk 14]
```diff
@@ -0,0 +1,28 @@
...
+
+class MoviesList extends Component {
+    componentDidMount(){
+        this.props.sortTitles();
+    }
+
+    render() {
+        const {data} = this.props;        
+        return data.map(movie => {
+            return (
```
Comment (byslavik) [inline src/components/MoviesList.js]: you can avoid that 'return'

File: src/components/common/Button.js [hunk 15]
```diff
@@ -0,0 +1,9 @@
+import React from 'react'
+const Button = ({ text, ...props }) => {
+    return ( 
+        <button type="button" {...props}>{text}</button>
```
Comment (byslavik) [inline src/components/common/Button.js]: why you are not using children?

Comment (NJaryal) [reply src/components/common/Button.js]: Updated - Using Now

File: src/components/Main.js (hunk 13 above)
Comment (NJaryal) [reply src/components/Main.js]: Functional Component Updated

File: src/components/MoviesList.js (hunk 14 above)
Comment (NJaryal) [reply src/components/MoviesList.js]: Updated

File: src/components/NotFound.js (hunk 4 above)
Comment (NJaryal) [reply src/components/NotFound.js]: Implemented

File: src/components/Search.js (hunk 8 above)
Comment (NJaryal) [reply src/components/Search.js]: Very Useful - Updated

File: src/components/Header.js [hunk 16]
```diff
@@ -0,0 +1,111 @@
...
+import MoviesList from './MoviesList'
+import MoviesData from '../data/moviesList'
+import Search from './Search'
+import SortSearch from './SortSearch'
+import SortByCategory from './SortByCategory'
+
+class Header extends Component {
+    constructor(props) {
+        super(props);
+        this.MoviesData = MoviesData.data;
```
Comment (byslavik) [inline src/components/Header.js]: better to put such kind of data to state

Comment (byslavik) [reply src/components/Header.js]: and do not keep it in class

File: src/components/MoviesList.js [hunk 17]
```diff
@@ -0,0 +1,55 @@
...
+    render() {
+        const {data} = this.props;
+        const MovieList = () => data.map(movie => 
+            <div className="card col-md-3 mx-3 my-3 px-0" key={movie.id}>
+                <img className="card-img-top" src={movie.poster_path} onClick={this.detailedMovieAPI.bind(null, movie.id)} alt={movie.title} />
+                <div className="card-body">                    
+                    <h5 className="card-title">{movie.title}</h5>
+                    <p className="card-text clearfix">
+                    <span className="float-left">{movie.title}</span> 
+                    <button type="button" className="btn btn-outline-secondary btn-sm float-right">{movie.release_date}</button>
```
Comment (byslavik) [inline src/components/MoviesList.js]: Please use common component here

File: src/components/MoviesList.js [hunk 18]
```diff
@@ -0,0 +1,55 @@
...
+            singleMovieData:"",
+            isDetailedView: false
+        }
+    }
+    componentDidMount(){
+        this.props.sortTitles();        
+    }   
+
+    detailedMovieAPI =(id) =>{
+        fetch(`https://reactjs-cdp.herokuapp.com/movies/${id}`)
```
Comment (byslavik) [inline src/components/MoviesList.js]: Please create service api function for that and put it into services folder

File: src/components/Header.js (hunk 16 above)
Comment (NJaryal) [reply src/components/Header.js]: Updated

File: src/components/MoviesList.js (hunk 18 above)
Comment (NJaryal) [reply src/components/MoviesList.js]: Created + Updated

File: src/components/MovieCard.js [hunk 19]
```diff
@@ -0,0 +1,11 @@
+import React from 'react'
+const MovieCard = ({title, release_date, genres}) => 
+        <div className="card-body">                    
+            <h5 className="card-title">{title}</h5>
+            <p className="card-text clearfix">
+                <span className="float-left">{title}</span> 
+                <button type="button" className="btn btn-outline-secondary btn-sm float-right">{release_date}</button>
```
Comment (byslavik) [inline src/components/MovieCard.js]: Why release date is a button?

Comment (NJaryal) [reply src/components/MovieCard.js]: Updated it with Span tag

**Pull request merged**